File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution 
{
public:
    int searchInsert(vector<int>& nums, int target) 
    {
        //Initialize the start and end position
        int start =0;
        int end = nums.size()-1;
        //use a binary Search
            while(start<=end)
            {
                int mid = start+(end - start)/2;
                if(nums[mid]==target)
                {
                    return mid;
                }
                else if(nums[mid]<target)
                {
                    start = mid +1;
                }
                else
                {
                    end = mid -1;
                }
            }
        //returning the start postion where we want to insert thae element
        return start;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
    
    // If the current element is greater than or equal to the target, return its index
            if (nums[i] >= target) {
                return i;
            }
        }
        //return the length of the array means no element is greater or equal
        return nums.length;
    }
}
